core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedRegionPlanGenerator.scala

```diff
@@ -85,7 +87,12 @@ class CostBasedRegionPlanGenerator(
   private def tryConnectRegionDAG(
       matEdges: Set[PhysicalLink]
   ): Either[DirectedAcyclicGraph[Region, RegionLink], DirectedPseudograph[Region, RegionLink]] = {
-    val regionDAG = new DirectedAcyclicGraph[Region, RegionLink](classOf[RegionLink])
+    val regionDAG = new DirectedAcyclicGraph[Region, RegionLink](
+      null, // vertexSupplier
+      SupplierUtil.createSupplier(classOf[RegionLink]), // edgeSupplier
+      false, // weighted
+      true // allowMultipleEdges
```

Comment:
can we add a test case to show that this can handle parallel edges correctly?

Reply:
Sure, I can add it in another PR where I will add test cases for CostBasedRegionPlanGenerator.

Reply:
Can you add it here? The review also would require test case to know its functionality

Reply:
Added a test case. Also updated all places where a region plan is used to support parallel edges.